metadata.json

```diff
@@ -40,6 +40,12 @@
       ]
     }
   ],
+  "requirements": [
+    {
+      "name": "puppet",
+      "version_requirement": ">= 3.8.0 < 4.0.0"
```

Comment:
Also, does it have an impact on acceptance testing, where the tests run `puppet agent` after upgrading?